bot/cogs/help_channels.py

```diff
@@ -0,0 +1,671 @@
+import asyncio
+import bisect
+import inspect
+import json
+import logging
+import random
+import typing as t
+from collections import deque
+from datetime import datetime
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+
+from bot import constants
+from bot.bot import Bot
+from bot.decorators import with_role
+from bot.utils.scheduling import Scheduler
+
+log = logging.getLogger(__name__)
+
+ASKING_GUIDE_URL = "https://pythondiscord.com/pages/asking-good-questions/"
+MAX_CHANNELS_PER_CATEGORY = 50
+
+AVAILABLE_TOPIC = """
+This channel is available. Feel free to ask a question in order to claim this channel!
+"""
+
+IN_USE_TOPIC = """
+This channel is currently in use. If you'd like to discuss a different problem, please claim a new \
+channel from the Help: Available category.
+"""
+
+DORMANT_TOPIC = """
+This channel is temporarily archived. If you'd like to ask a question, please use one of the \
+channels in the Help: Available category.
+"""
+
+AVAILABLE_MSG = f"""
+This help channel is now **available**, which means that you can claim it by simply typing your \
+question into it. Once claimed, the channel will move into the **Help: In Use** category, and will \
+be yours until it has been inactive for {constants.HelpChannels.idle_minutes} minutes. When that \
+happens, it will be set to **dormant** and moved into the **Help: Dormant** category.
+
+You may claim a new channel once every {constants.HelpChannels.claim_minutes} minutes. If you \
+currently cannot send a message in this channel, it means you are on cooldown and need to wait.
+
+Try to write the best question you can by providing a detailed description and telling us what \
+you've tried already. For more information on asking a good question, \
+[check out our guide on asking good questions]({ASKING_GUIDE_URL}).
```

Comment:
Typically, embedded links limit the text to just noun, and not the entire subject. I.E. "guide on asking good questions" or just "asking good questions" instead of "check out our guide on asking good questions".

Reply:
I don't mind either way, although I think the noun-only version looks slightly better.

Reply:
As per @aeros 's suggestion, it would look like this:
```suggestion
check out our guide on [asking good questions]({ASKING_GUIDE_URL}).
```